### ISSUE TITLE
How to fully ignore links into markdown

### ISSUE THREAD

**Issue by sw360cab** [jdkato/txtlint#1021]
I am using Vale for validating Markdown files.
I want to exclude from Vale's validation the references I add at the bottom as bullet list of md links.

I have read #320, and I understood Vale is not validating the url part (brackets), but in my case I want to ignore also the content of square brackets.
I tried many methods but the error from Vale is still there.

In particular I have tried writing a regular expressions

```
TokenIgnores = (\[.*\]\(.*\))
```

For regex101 it is a valid [Group1 regexp](https://regex101.com/r/mFM0kZ/1/)
Can you suggest me what is the correct way to achieve that?

**Comment by napcs** [comment]
@sw360cab This isn't official, but one thing I've done for these situations is been explicit in my docs and use comments in the docs to control Vale.

If these are Markdown docs you can use `vale off`:

```
<!-- vale off -->

Links:
- [foobar](https://example.com)
- [blablablah](https://example.com)
...

<!-- vale on -->

```

You can also get specific with the rule you want to turn off:

```
<!-- vale Vale.Terms = NO -->
...

<!-- vale Vale.Terms = YES -->

```

That's how I've managed to do this. It's a bit of work to add the markup but it's nice that it's also clear that you're choosing to ignore sections of the content this way.

**Comment by jdkato** [comment]
Your regex is fine — the problem is where it's placed. `TokenIgnores` is syntax-specific, so it has to go inside a format section. In the global section Vale rejects it:

```
.vale.ini:3:E201:'TokenIgnores' is a syntax-specific option
```

Move it under `[*.md]`:

```ini
StylesPath = styles
MinAlertLevel = suggestion

[*.md]
BasedOnStyles = Vale
TokenIgnores = (\[.*\]\(.*\))
```

Given:

```markdown
Some intro with a mispeled word.

- [teh mispeled linktext](https://example.com/teh)
- [another baddword here](https://example.com/x)

Trailing badspelling outside.
```

that reports only the two errors outside the links:

```
t.md:1:19:Vale.Spelling:Did you really mean 'mispeled'?
t.md:6:10:Vale.Spelling:Did you really mean 'badspelling'?
```

Both the link text and the URL are skipped.
